### PR TITLE
Prepare for MELPA packaging

### PR DESCRIPTION
**Pull request by svmhdvn**
This change attempts to handle all feedback from the MELPA packaging request, initiated here: https://github.com/melpa/melpa/pull/8239#issuecomment-1295922381. This is the current list of changes I've done at a high level:

* Rename functions to use their containing package's prefixes. This fixes a lot of the `package-lint` feedback
* Rename package file names to better suit their purpose and to be more consistent with the already available function prefixes inside them
* Add missing doc strings to functions and variables
* Add missing package metadata sections
* Update the version string to 0.1.0 to match `bqn-mode-pkg.el`
* Fix minor grammatical and spacing issues
* Fix a bug in `bqn-help-symbol-at-point-is-called` where a `(when ...)` call had an empty body
* Rename license filename to more common standard and added SPDX identifiers everywhere

Please let me know if any changes are necessary. Particularly, if any of the renaming or added doc strings are incorrect.

### CODE REVIEW

File: bqn-keymap-mode.el [hunk 1]
```diff
@@ -1,29 +1,35 @@
-;;; bqn-keyboard --- BQN Keyboard Reference
-;;;
+;;; bqn-keymap-mode.el --- BQN Keyboard Reference
```
Comment (AndersonTorres) [inline bqn-keymap-mode.el]:
```suggestion
;;; bqn-keymap-mode.el --- BQN Keymap Reference
```

File: bqn-keymap-mode.el [hunk 2]
```diff
@@ -1,29 +1,35 @@
...
+;; Author: Marshall Lochbaum <mwlochbaum@gmail.com>
+;; Version: 0.1.0
+;; Package-Requires: ((emacs "24.3"))
+;; URL: https://github.com/museoa/bqn-mode
+;; SPDX-License-Identifier: GPL-3.0-or-later
+
 ;;; Commentary:
-;;;
+
+;; This package provides a major mode for displaying the BQN keyboard reference.
```
Comment (AndersonTorres) [inline bqn-keymap-mode.el]:
```suggestion
;; This package provides a major mode for displaying the BQN keymap reference.
```

File: bqn-mode.el [hunk 3]
```diff
@@ -5,9 +5,10 @@
 ;; Changes are copyright 2021 Marshall Lochbaum <mwlochbaum@gmail.com>.
 
 ;; Author: Marshall Lochbaum <mwlochbaum@gmail.com>
-;; Version: 0.0.0
+;; Version: 0.1.0
```
Comment (AndersonTorres) [inline bqn-mode.el:8]: Why?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: bqn-mode.el (hunk 3 above)
Comment (svmhdvn) [reply bqn-mode.el:8]: This matches the version number from `bqn-mode-pkg.el`

**Pull request merged**